app/controllers/GiftAidSubmissionController.scala

```diff
@@ -34,7 +34,7 @@ class GiftAidSubmissionController @Inject()(
                                            )(implicit ec: ExecutionContext) extends BackendController(cc) {
 
   def submit(nino: String, taxYear: Int): Action[AnyContent] = auth.async { user =>
-    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(user.headers, Some(user.session))
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(user, user.session)
```

Comment:
SHould this be HeaderCarrierConverter.fromRequest(request) like the other services?